src/test/java/org/openrewrite/java/spring/data/MigrateAuditorAwareToOptionalTest.java

```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.spring.data;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.Issue;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+@Issue("https://github.com/openrewrite/rewrite-spring/issues/613")
+class MigrateAuditorAwareToOptionalTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec
+          .parser(JavaParser.fromJavaVersion().classpath("spring-data-commons"))
+          .recipe(new MigrateAuditorAwareToOptional());
+    }
+
+    @DocumentExample
+    @Test
+    void rewriteImplementation() {
+        //TODO Question for TIM: how to get rid of the types? I have the imports.
+        //-    public Optional<String> getCurrentAuditor() {
+        //+    public java.util.Optional<java.lang.String> getCurrentAuditor() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              package sample;
+              
+              import org.springframework.data.domain.AuditorAware;
+              
+              public class MyAuditorAware implements AuditorAware<String> {
+                  @Override
+                  public String getCurrentAuditor() {
+                      return "admin";
```

Comment:
Most of the tests seem to use String literals; would it make sense have a unit test a little more reflective of expected use, like method invocations [seen here](https://github.com/timtebeek/spring-security-samples/blob/2e6d81211abb8ddae5589e41c2bf5d477143b2d7/audit-spring-data-entities/src/main/java/com/jdriven/AuditSecurityConfiguration.java#L20-L27)?